admin-dev/themes/new-theme/js/components/modal/progress-modal.ts

```diff
@@ -0,0 +1,421 @@
+    );
+    this.progressPercentage.append('0%');
+
+    this.body = document.createElement('div');
+    this.body.classList.add('modal-body', 'text-left', 'font-weight-normal');
+
+    this.body.append(this.progressPercentage);
+    this.body.append(this.buildProgressBar());
+
+    this.switchToErrorButton = document.createElement('div');
+    this.switchToErrorButton.classList.add(
+      'switch-to-errors-button',
+      'alert',
+      'alert-warning'
+    );
+
+    this.errorCount = document.createElement('span');
+    this.errorCount.classList.add(
+      'modal-error-count'
+    )
+    this.errorCount.innerHTML = '0';
+
+    let errorLogString = 'View %d error logs';
+    this.switchToErrorButton.innerHTML = errorLogString.replace('%d', this.errorCount.outerHTML);
+
+    this.header.append(this.switchToErrorButton);
+
+    let errorContainer = document.createElement('div');
+    errorContainer.classList.add('modal-error-container-single');
+    this.body.append(errorContainer);
+    // Modal footer element
+    this.footer = document.createElement('div');
+    this.footer.classList.add('modal-footer');
+
+    this.stopProcessingButton = document.createElement('button');
+    this.stopProcessingButton.setAttribute('type', 'button');
+    this.stopProcessingButton.classList.add('btn', 'btn-outline-secondary', 'btn-lg', 'stop-processing');
+    this.stopProcessingButton.innerHTML = 'Stop processing';
+
+    // Appending element to the modal
+    this.footer.append(this.stopProcessingButton, ...params.customButtons);
+    this.content.append(this.footer);
+
+
+    // Modal title element
+    if (params.modalTitle) {
+      this.title = document.createElement('h4');
+      this.title.classList.add('modal-title');
+      this.title.innerHTML = params.modalTitle;
+    }
+
+    // Constructing the modal
+    if (this.title) {
+      this.header.appendChild(this.title);
+    }
+    this.content.append(this.header, this.body);
+  }
+
+  private buildProgressBar(): HTMLElement
+  {
+    let progress = document.createElement('div');
+    progress.setAttribute('style', 'display: block; width: 100%');
+
+    progress.classList.add
+    (
+      'progress',
+      'active',
+    );
+    let progressBar = document.createElement('div');
+    progressBar.classList.add('progress-bar', 'progress-bar-success');
+    progressBar.setAttribute('role', 'progressbar');
+    progressBar.setAttribute('style', 'width: 0%;');
+
+    this.progressDone = document.createElement('div');
+    this.progressDone.setAttribute('style', 'width: 0%');
+
+    this.progressDone.classList.add
+    (
+      'progress-bar',
+      'progress-bar-success',
+    );
+    this.progressDone.setAttribute('role', 'progressbar');
+    this.progressDone.setAttribute('id', 'modal_progressbar_done');
+    progress.append(this.progressDone);
+
+    return progress;
+  }
+}
+export class ProgressModalErrorContainer {
+  footer!: HTMLElement;
+
+  container!: HTMLElement;
+
+  dialog!: HTMLElement;
```

Comment:
These two elements are not useful here since they are handled by the parent container

Reply:
```suggestion
```